src/lib.rs

```diff
@@ -273,12 +276,37 @@ where
     Ok(state)
 }
 
+/// The error returned by [`cmark_resume_one_event`] and
+/// [`cmark_resume_with_source_range_and_options`].
+#[derive(Debug)]
+pub enum Error {
+    FormatFailed(fmt::Error),
+    UnexpectedEvent,
+}
+
+impl fmt::Display for Error {
```

Comment:
Depending on the policy for new dependencies in pulldown-cmark-to-cmark, I think you could consider using thiserror here. It will basically expand to the hand-written code here.

See https://docs.rs/thiserror/latest/thiserror/.

Reply:
I'd want to avoid adding dependencies here.
In general, I want to use `thiserror` less, not more, but if there was a library with a lot of error handling, right now there would be no way around it.